### ISSUE TITLE
Product search returns "Return data was in an unexpected format."

### ISSUE THREAD

**Issue by dingman** [trafficgate/linkshare-api#11]
Hi! Thanks for such a great package. I've started integrating it into my Laravel application and running into some trouble.

Here is the Gist of my vendor file: https://gist.github.com/dingman/5b6f022b266e2d6a457654dcd36f0681

It seems to be setting up the token successfully and calling the URL, however I get an error each time.

```
Making one call every 6 second(s) to obey the limit of 10 requests per minute.
API URL:                https://api.rakutenmarketing.com/productsearch/1.0?keyword=Jill+Jill+Stuart&max=100&pagenumber=1&mid=3184&sort=retailprice&sorttype=asc


In ProductSearch.php line 76:

  Return data was in an unexpected format.
```

Any help is appreciated, TIA!

**Comment by ikari7789** [comment]
First of all, thank you for the compliment!

Secondly, I think this may be related to #10.

It seems that some kind of backwards breaking change may have been added to Guzzle at some point that this package didn’t properly handle, and I’m sorry about that. It’s working for us in production, but we haven’t updated it in quite a while and I think that Guzzle, at some point, made its PSR7 response the default.

I’ll see what I can do to fix that.

**Comment by dingman** [comment]
Very well could be. I'm using a normal Guzzle\Client() in another part of my code, doesn't seem to be causing any issue.

$result ends up being an object with the current implementation

```
vagrant@homestead:~/code$ php artisan macys:search
Making one call every 6 second(s) to obey the limit of 10 requests per minute.
API URL:                https://api.rakutenmarketing.com/productsearch/1.0?keyword=Jill+Jill+Stuart&max=100&pagenumber=1&mid=3184&sort=retailprice&sorttype=asc

GuzzleHttp\Psr7\Response {#765
  -reasonPhrase: "OK"
  -statusCode: 200
  -headers: array:9 [
    "Access-Control-Allow-Headers" => array:1 [
      0 => "authorization,Access-Control-Allow-Origin,Content-Type"
    ]
    "Access-Control-Allow-Methods" => array:1 [
      0 => "GET,POST,PUT,DELETE,OPTIONS"
    ]
    "Content-Type" => array:1 [
      0 => "application/xml; charset=UTF-8"
    ]
    "Date" => array:1 [
      0 => "Tue, 27 Feb 2018 17:41:14 GMT"
    ]
    "P3P" => array:1 [
      0 => "CP="CAO DSP COR CURa ADMa DEVa TAIa PSAa PSDa IVAa IVDa CONa HISa OTPa OUR IND UNI NAV OTC""
    ]
    "Server" => array:1 [
      0 => "PopShops"
    ]
    "Vary" => array:1 [
      0 => "Accept-Encoding"
    ]
    "transfer-encoding" => array:1 [
      0 => "chunked"
    ]
    "Connection" => array:1 [
      0 => "keep-alive"
    ]
  ]
  -headerNames: array:9 [
    "access-control-allow-headers" => "Access-Control-Allow-Headers"
    "access-control-allow-methods" => "Access-Control-Allow-Methods"
    "content-type" => "Content-Type"
    "date" => "Date"
    "p3p" => "P3P"
    "server" => "Server"
    "vary" => "Vary"
    "transfer-encoding" => "transfer-encoding"
    "connection" => "Connection"
  ]
  -protocol: "1.1"
  -stream: GuzzleHttp\Psr7\Stream {#764
    -stream: stream resource {@470
      wrapper_type: "PHP"
      stream_type: "TEMP"
      mode: "w+b"
      unread_bytes: 0
      seekable: true
      uri: "php://temp"
      options: []
    }
    -size: null
    -seekable: true
    -readable: true
    -writable: true
    -uri: "php://temp"
    -customMetadata: []
  }
}
```

...with a Psr7 response.

Anything you can do is greatly appreciated :)

**Comment by dingman** [comment]
@ikari7789 appreciate anything you can do, any chance you've been able to look at this? I'm eager to get integrated with the Linkshare API :) TIA!

**Comment by ikari7789** [comment]
Hi @dingman this should be fixed by the latest release v1.0.9.

**Comment by dingman** [comment]
Thank you Andrew! This worked great. A few tweaks since echo $result was resulting in "cannot convert to string", so I ```return $result``` and that did the trick.

Cheers!
